packages/wonder-blocks-tooltip/util/is-obscured.test.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+import {View} from "@khanacademy/wonder-blocks-core";
+
+import {mount} from "enzyme";
+import isObscured from "./is-obscured.js";
+
+describe("isObscured", () => {
+    test("element is null, throws", () => {
+        // Arrange
+        const element = ((null: any): Element);
+
+        // Act
+        const underTest = () => isObscured(element);
+
+        // Assert
+        expect(underTest).toThrowError();
+    });
+
+    test("element is not obscured by anything, returns false", (done) => {
+        const arrange = (actAndAssert) => {
+            // Arrange
+            const nodes = (
+                <View>
+                    <View ref={actAndAssert}>Unobscured</View>
+                </View>
+            );
+            mount(nodes);
+        };
+
+        const actAndAssert = (ref) => {
+            const element = ((ReactDOM.findDOMNode(ref): any): Element);
+
+            const ogElementFromPoint = document.elementFromPoint;
+            // When not obscurred, elementFromPoint should return the element.
+            // Flow doesn't like us doing this to the document $FlowFixMe
+            document.elementFromPoint = jest.fn().mockReturnValue(element);
+
+            // Act
+            const result = isObscured(element);
+
+            // Assert
+            expect(result).toBeFalsy();
+            // Reset the document method to avoid side-effects.
+            // Flow doesn't like us doing this to the document $FlowFixMe
+            document.elementFromPoint = ogElementFromPoint;
```

Comment:
I'm a little shaky with how `beforeEach` and `afterEach` interact with async tests so I'm cool with handling the reset inside each `test`.

Reply:
Yes, I considered the before each and after each. I think they work fine for async (I think the after each only runs once done() or timeout occurs), but I wasn't sure a single line in each warranted adding them. I could go either way.

Reply:
I've added them. I think it tidies things up considerably and appears to work.